src/main/java/application/repository/builders/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,35 @@
+package application.repository.builders;
+
+import application.dto.BookSearchParametersDto;
+import application.model.Book;
+import application.repository.SpecificationBuilder;
+import application.repository.SpecificationProviderManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+
+    private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParametersDto bookSearchParametersDto) {
+        Specification<Book> specification = Specification.where(null);
+        if (bookSearchParametersDto.authors() != null
+                && bookSearchParametersDto.authors().length > 0) {
+            specification = specification.and(bookSpecificationProviderManager
+                    .getSpecificationProvider("author")
+                    .getSpecification(bookSearchParametersDto.authors()));
+        }
+        if (bookSearchParametersDto.titles() != null
+                && bookSearchParametersDto.titles().length > 0) {
+            specification = specification.and(bookSpecificationProviderManager
+                    .getSpecificationProvider("title")
+                    .getSpecification(bookSearchParametersDto.titles()));
+        }

```

Comment:
Could we somehow split that into dedicated private method, each per each parameter?